File: app/src/main/java/se/gustavkarlsson/skylight/android/util/CountdownTimer.java
package se.gustavkarlsson.skylight.android.util;

import org.threeten.bp.Clock;
import org.threeten.bp.Duration;
import org.threeten.bp.Instant;

public class CountdownTimer {
	private final Instant expiryTime;
	private final Clock clock;

	public CountdownTimer(Duration duration, Clock clock) {
		this.expiryTime = clock.instant().plus(duration);
		this.clock = clock;
	}

	public Duration getRemainingTime() {
		Instant now = clock.instant();
		return Duration.between(now, expiryTime);
	}

}

File: app/src/main/java/se/gustavkarlsson/skylight/android/util/CountdownTimer.kt
package se.gustavkarlsson.skylight.android.util

import org.threeten.bp.Clock
import org.threeten.bp.Duration
import se.gustavkarlsson.skylight.android.extensions.until

internal class CountdownTimer(
		duration: Duration,
		private val clock: Clock
) {
	private val expiryTime = clock.instant().plus(duration)

	val remainingTime: Duration
		get() {
			val now = clock.instant()
			return now.until(expiryTime)
		}

}
